isce2_topsapp/iono_proc.py

```diff
@@ -0,0 +1,383 @@
+    # for [ymin, ymax, xmin, xmax]
+    extent_isce = [extent[k] for k in [1, 3, 0, 2]]
+
+    # Update PATH with ISCE2 applications
+    isce_application_path = Path(f'{site.getsitepackages()[0]}'
+                                    '/isce/applications/')
+    os.environ['PATH'] += (':' + str(isce_application_path))
+
+    # Define topsApp input xml parameters
+    iono_kwargs = {'orbit_directory': orbit_directory,
+                   'output_reference_directory': 'reference',
+                   'output_secondary_directory': 'secondary',
+                   'ref_zip_file': reference_slc_zips,
+                   'sec_zip_file': secondary_slc_zips,
+                   'region_of_interest': extent_isce,
+                   'demFilename': dem_for_proc,
+                   'geocodeDemFilename': dem_for_geoc,
+                   'filter_strength': .5,
+                    'do_unwrap': True,
+                    'use_virtual_files': True,
+                    'do_esd': do_esd,
+                    'esd_coherence_threshold': esd_coherence_threshold,
+                    # IONO PARAMETERS
+                    'estimate_ionosphere_delay': True,
+                    'iono_burstProperties': False,
+                    'iono_polyFit': True,
+                    'iono_correctAzimuthshift': 1,
+                    'azimuth_looks': azimuth_looks,
+                    'range_looks': range_looks,
+                    'swaths': swaths,
+                    'geocode_list': GEOCODE_LIST_BASE
+                     }
+
+    with open(TEMPLATE_DIR/'topsapp_iono_template.xml', 'r') as file:
+        template = Template(file.read())
+
+    # Step-1 Run ionospheric correction step subband
+    iono_xml = template.render(**iono_kwargs,
+                               iono_startStep=IONO_STEPS[0],
+                               iono_stopStep=IONO_STEPS[0])
+
+    with open('ionoApp.xml', "w") as file:
+        file.write(iono_xml)
+
+    tops_app_cmd = f'{isce_application_path}/topsApp.py'
+
+    step_cmd = f'{tops_app_cmd} ionoApp.xml --dostep=ion'
+    result = subprocess.run(step_cmd,
+                            shell=True)
+    if result.returncode != 0:
+        raise ValueError('TopsApp failed at step: ion-subband')
+
+    # Step-2 Mask upper and lower band burst wrapped interferograms
+    # path to burst ifgs: ion/lower|upper/fine_interferogram/IW{1,2,3}/burst_{##}.int
+    # path to burst geometry: geom_reference/IW{1,2,3}/lat|lon_{##}.rdr
+
+    if mask_filename:
+        # Project mask to burst image coordinate space
+        workdir = Path('.').absolute()
+
+        get_swath = lambda x: x.split('/')[-2]
+        get_burst = lambda x: x.split('/')[-1].split('_')[-1].split('.')[0]
+
+        # Get all geometry files
+        geom_files = list(workdir.glob('geom_reference/IW*/lat*.rdr'))
+
+        # NOTE: THis can be easily parallized, skipped for now
+        with tqdm(total=len(geom_files)) as pbar:
+            for lat in geom_files:
```

Comment:
Why not just use something like:

```
from tqdm import tqdm

for lat in tqdm(geom_files):
```
Simplifies a little.

Reply:
I made it like this to be able to update description of progress bar, probably there is nicer solution than this one